iceoryx2/src/config.rs

```diff
@@ -71,19 +71,28 @@
 
 use core::time::Duration;
 use iceoryx2_bb_container::semantic_string::SemanticString;
-use iceoryx2_bb_elementary::lazy_singleton::*;
-use iceoryx2_bb_posix::{file::FileBuilder, shared_memory::AccessMode};
+use iceoryx2_bb_elementary::{lazy_singleton::*, CallbackProgression};
+use iceoryx2_bb_posix::{
+    file::FileBuilder, shared_memory::AccessMode, system_configuration::get_global_config_path,
+};
 use iceoryx2_bb_system_types::file_name::FileName;
 use iceoryx2_bb_system_types::file_path::FilePath;
 use iceoryx2_bb_system_types::path::Path;
 use serde::{Deserialize, Serialize};
 
-use iceoryx2_bb_log::{debug, fail, trace, warn};
+use iceoryx2_bb_log::{fail, fatal_panic, trace, warn};
 
 use crate::service::port_factory::publisher::UnableToDeliverStrategy;
 
-/// Path to the default config file
-pub const DEFAULT_CONFIG_FILE: &[u8] = b"config/iceoryx2.toml";
+const DEFAULT_CONFIG_FILE_NAME: &[u8] = b"iceoryx2.toml";
+const DEFAULT_CONFIG_PATH: &[u8] = b"config";
```

Comment:
I don't understand the naming of `DEFAULT_CONFIG_PATH`, especially with the existence of `RELATIVE_CONFIG_FILE_PATH`

Edit. okay, with the context below, I understand it, but I would just remove this. Alternatively, please consider naming this `RELATIVE_LOCAL_CONFIG_PATH`